testing/src/test/java/org/oppia/android/testing/lightweightcheckpointing/ExplorationCheckpointTestHelperTest.kt

```diff
@@ -79,20 +89,77 @@ class ExplorationCheckpointTestHelperTest {
   @Before
   fun setUp() {
     setUpTestApplicationComponent()
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
   }
 
   private fun setUpTestApplicationComponent() {
     ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
   }
 
   @Test
-  fun testSaveFakeExplorationCheckpoint_checkCheckpointIsSaved() {
-    explorationCheckpointTestHelper.saveFakeExplorationCheckpoint(profileId.internalId)
+  fun testSaveCheckpointForFractionsStory0Exploration0_checkCheckpointIsSaved() {
+    explorationCheckpointTestHelper.saveCheckpointForFractionsStory0Exploration0(
+      profileId = profileId,
+      version = FRACTIONS_STORY_0_EXPLORATION_0_CORRECT_VERSION
+    )
+
+    val retrieveFakeCheckpointLiveData =
+      explorationCheckpointController.retrieveExplorationCheckpoint(
+        profileId,
+        FRACTIONS_EXPLORATION_ID_0
+      ).toLiveData()
+    retrieveFakeCheckpointLiveData.observeForever(mockExplorationCheckpointObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    // Verify exploration checkpoint was saved with correct exploration title.
+    verify(mockExplorationCheckpointObserver, atLeastOnce())
+      .onChanged(explorationCheckpointCaptor.capture())
+    assertThat(explorationCheckpointCaptor.value.isSuccess()).isTrue()
+    assertThat(explorationCheckpointCaptor.value.getOrThrow().explorationTitle)
+      .isEqualTo(FRACTIONS_EXPLORATION_0_TITLE)
+  }
+
+  @Test
+  fun testUpdateCheckpointForFractionsStory0Exploration0_checkCheckpointIsSaved() {
+    explorationCheckpointTestHelper.saveCheckpointForFractionsStory0Exploration0(
+      profileId = profileId,
+      version = FRACTIONS_STORY_0_EXPLORATION_0_CORRECT_VERSION
+    )
+
+    explorationCheckpointTestHelper.updateCheckpointForFractionsStory0Exploration0(
+      profileId = profileId,
+      version = FRACTIONS_STORY_0_EXPLORATION_0_CORRECT_VERSION
+    )
+
+    val retrieveFakeCheckpointLiveData =
+      explorationCheckpointController.retrieveExplorationCheckpoint(
+        profileId,
+        FRACTIONS_EXPLORATION_ID_0
+      ).toLiveData()
+    retrieveFakeCheckpointLiveData.observeForever(mockExplorationCheckpointObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    // Verify exploration checkpoint was saved with correct exploration title.
+    verify(mockExplorationCheckpointObserver, atLeastOnce())
+      .onChanged(explorationCheckpointCaptor.capture())
+    assertThat(explorationCheckpointCaptor.value.isSuccess()).isTrue()
+    assertThat(explorationCheckpointCaptor.value.getOrThrow().explorationTitle)
+      .isEqualTo(FRACTIONS_EXPLORATION_0_TITLE)
+    assertThat(explorationCheckpointCaptor.value.getOrThrow().pendingStateName)
+      .isEqualTo(FRACTIONS_STORY_0_EXPLORATION_0_SECOND_STATE_NAME)
```

Comment:
Could you also add a verification line for state name in the previous test? Otherwise it seems a bit like update*() could do nothing & both tests would still pass (it's not obvious what the "difference" state being verified here is).

Ditto for the other similar tests here.

Reply:
done